### PR TITLE
[Docs] Japanese translation of docs/reference_keymap_extras.md

### PR DESCRIPTION
**Pull request by umi-umi**
## Description

Add Japanese translation of docs/reference_keymap_extras.md.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by s-show — APPROVED** [review]

File: docs/ja/reference_keymap_extras.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  original document: 0.9.55:docs/reference_keymap_extras.md
+  git diff 0.9.55 HEAD -- docs/reference_keymap_extras.md | cat
+-->
+
+キーボードは多くの言語をサポートすることができます。ただし、それらはキーを押したことで生成される実際の文字を送信しません - 代わりに数字のコードを送信します。USB HID の仕様ではそれらは "usages" と呼ばれますが、キーボードの文脈では「スキャンコード」あるいは「キーコード」として参照されることが多いです。
+HID Keyboard/Keypad usage ページでは 256 未満の usage が定義されており、それらの一部は最新のオペレーティングシステムでは機能しません。では、この言語のサポートはどのようにして実現されるのでしょうか？
+
+簡単に言うと、オペレーティングシステムはユーザが設定したキーボードレイアウトに基づいて受け取った usage を適切な文字にマップします。例えば、スウェーデン人がキーボードに印刷されている `å` 文字のキーを押すと、キーボードは *実際には* `[` のキーコードを送信します。
+
+明らかにこれは混乱する可能性があるため、QMK は多くのキーボードレイアウトのために言語固有のキーコードのエイリアスを提供します。これらは、それだけではあまり多くの事を行いません - OS の設定で対応するキーボードレイアウトをまだまだ設定する必要があります。それらをキーマップのキーキャップラベルと考えてください。
```
Comment (mtei) [inline docs/ja/reference_keymap_extras.md:13]:
```suggestion
明らかにこれは混乱する可能性があるため、QMK は多くのキーボードレイアウトのために言語固有のキーコードのエイリアスを提供します。これらはそれだけでは何もしません - さらに OS の設定で対応するキーボードレイアウトを設定する必要があります。それらをキーマップのキーキャップラベルと考えてください。
```

**Review by mtei — APPROVED** [review]

File: docs/ja/reference_keymap_extras.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# 言語固有のキーコード
+
+<!---
+  original document: 0.9.55:docs/reference_keymap_extras.md
+  git diff 0.9.55 HEAD -- docs/reference_keymap_extras.md | cat
+-->
+
+キーボードは多くの言語をサポートすることができます。ただし、それらはキーを押したことで生成される実際の文字を送信しません - 代わりに数字のコードを送信します。USB HID の仕様ではそれらは "usages" と呼ばれますが、キーボードの文脈では「スキャンコード」あるいは「キーコード」と呼ばれることが多いです。
+HID Keyboard/Keypad usage ページでは 256 未満の usage が定義されており、それらの一部は最新のオペレーティングシステムでは機能しません。では、この言語のサポートはどのようにして実現されるのでしょうか？
```
Comment (shelaf) [inline docs/ja/reference_keymap_extras.md:9]: 最新、は言い過ぎかと。
```suggestion
HID Keyboard/Keypad usage ページでは 256 未満の usage が定義されており、それらの一部は現在のオペレーティングシステムでは機能しません。では、この言語のサポートはどのようにして実現されるのでしょうか？
```

File: docs/ja/reference_keymap_extras.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+簡単に言うと、オペレーティングシステムはユーザが設定したキーボードレイアウトに基づいて受け取った usage を適切な文字にマップします。例えば、スウェーデン人がキーボードに印刷されている `å` 文字のキーを押すと、キーボードは *実際には* `[` のキーコードを送信します。
```
Comment (shelaf) [inline docs/ja/reference_keymap_extras.md:11]:
```suggestion
簡単に言うと、オペレーティングシステムはユーザが設定したキーボードレイアウトに基づいて受け取った usage を適切な文字にマップします。例えば、スウェーデン人がキーボードの `å` という文字が刻印されたキーを押すと、キーボードは *実際には* `[` のキーコードを送信します。
```

File: docs/ja/reference_keymap_extras.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+| Plover (Dvorak)*    | `keymap_plover_dvorak.h` |
+| Steno*              | `keymap_steno.h`         |
+| Workman             | `keymap_workman.h`       |
+| Workman (ZXCVM)     | `keymap_workman_zxcvm.h` |
+
+## Sendstring サポート
+
+デフォルトでは、`SEND_STRING()` は US ANSI キーボードレイアウトが設定されたと見なします。別のレイアウトを使っている場合は、キーマップで(上記のように)`#include "sendstring_*.h"` して、ASCII 文字をキーコードにマッピングするために使われるルックアップテーブルを上書きすることができます。
+
+ここで注意すべき重要な点は、`SEND_STRING()` は [ASCII text](https://en.wikipedia.org/wiki/ASCII#Character_set) でのみ機能するということです。これは、ユニコード文字を含む文字列を渡すことができないことを意味します - 残念ながら、これには希望のレイアウトに存在するかもしれないアクセント付き文字が含まれています。
```
Comment (shelaf) [inline docs/ja/reference_keymap_extras.md:86]:
```suggestion
ここで注意すべき重要な点は、`SEND_STRING()` は [ASCII 文字](https://en.wikipedia.org/wiki/ASCII#Character_set) でのみ機能するということです。これは、ユニコード文字を含む文字列を渡すことができないことを意味します - 残念ながら、これには希望のレイアウトに存在する可能性のあるアクセント付き文字が含まれています。
```

File: docs/ja/reference_keymap_extras.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+多くのレイアウトでは、Grave または Tilde などの特定の文字を[デッドキー](https://en.wikipedia.org/wiki/Dead_key)としてのみ使えるようにしています。そのため、デッドキーが次の文字と潜在的に結合されることを防ぐためには、送信したい文字列の中のデッドキーのすぐ後にスペースを追加する必要があります。
+その他のレイアウトでは、ラテン語由来のアルファベット(例えば、Greek と Russian) を使わないため、Sendstring ヘッダーがありません。従って ASCII 文字セットのほとんどを入力する方法がありません。これらは上記の `*` でマークされています。
```
Comment (shelaf) [inline docs/ja/reference_keymap_extras.md:88]:
```suggestion
ラテン語由来のアルファベットを使わない(例えば、ギリシャ語やロシア語のような)他のレイアウトには、Sendstring ヘッダーがありません。従って ASCII 文字セットのほとんどを入力する方法がありません。これらは上記で `*` でマークされています。
```

Comment (umi-umi) [reply docs/ja/reference_keymap_extras.md]: ご指摘ありがとうございます。
訳を弄りまわしてもどこかすっきりしない感じがしていたのですが、すごく読みやすくなりました。
ありがとうございます。

**Review by shelaf — APPROVED** [review]

**Review by s-show — APPROVED** [review]

**Comment by mtei** [comment]
I think this is ready to merge.

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
ありがとうございました！

**Pull request merged**